### PR TITLE
Added Segment Anything interactor for GPU/CPU

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
Idea of the PR is to finish this one #5990
Deploy for GPU: ``./deploy_gpu.sh pytorch/facebookresearch/sam/nuclio/``
Deploy for CPU: ``./deploy_cpu.sh pytorch/facebookresearch/sam/nuclio/``

If you want to use GPU, be sure you setup docker for this [guide](https://github.com/NVIDIA/nvidia-docker/blob/master/README.md#quickstart).  

### How has this been tested?
Manual testing
demo2

Resolved issue #5984
But the interface probably can be improved

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: serverless/pytorch/facebookresearch/sam/nuclio/model_handler.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import numpy as np
```
Comment (nmanovic) [inline serverless/pytorch/facebookresearch/sam/nuclio/model_handler.py:1]: Need to add Copyright Header

File: serverless/pytorch/facebookresearch/sam/nuclio/main.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import json
```
Comment (nmanovic) [inline serverless/pytorch/facebookresearch/sam/nuclio/main.py:1]: Need to add copyright header

File: serverless/pytorch/facebookresearch/sam/nuclio/function.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+metadata:
```
Comment (nmanovic) [inline serverless/pytorch/facebookresearch/sam/nuclio/function.yaml:1]: Copyright header

File: serverless/pytorch/facebookresearch/sam/nuclio/function-gpu.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+metadata:
```
Comment (nmanovic) [inline serverless/pytorch/facebookresearch/sam/nuclio/function-gpu.yaml:1]: Copyright header

File: serverless/pytorch/facebookresearch/sam/nuclio/function-gpu.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  namespace: cvat
+  annotations:
+    name: Segment Anything
+    version: 2
+    type: interactor
+    spec:
+    framework: pytorch
+    min_pos_points: 1
+    min_neg_points: 0
+    help_message: The interactor allows to get a mask of an object using at least one positive, and any negative points inside it
```
Comment (nmanovic) [inline serverless/pytorch/facebookresearch/sam/nuclio/function-gpu.yaml:12]: Need to add gif from hrnet. Thus UI will show visual help how to work with the model.

File: serverless/pytorch/facebookresearch/sam/nuclio/function.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  namespace: cvat
+  annotations:
+    name: Segment Anything
+    version: 2
+    type: interactor
+    spec:
+    framework: pytorch
+    min_pos_points: 1
+    min_neg_points: 0
+    help_message: The interactor allows to get a mask of an object using at least one positive, and any negative points inside it
```
Comment (nmanovic) [inline serverless/pytorch/facebookresearch/sam/nuclio/function.yaml:12]: Need to add gif from hrnet. Thus UI will show visual help how to work with the model.

**Review by nmanovic — APPROVED** [review]

**Comment by medphisiker** [comment]
Thank you very much for integrating this neural network! Works like fBRs, but much more accurate. It's great that it has an inference on both CPU and GPU.

**Comment by davodogster** [comment]
Hi guys, really awesome work adding SAM as an interactor!! I have noticed it doesnt segment the border pixels sometimes, even if I select them as a positive label. Is this fixable?

image
This is via the online CVAT AI version. I haven't tested it in the self-hosted version.
Cheers, Sam
